contracts/factory.sol

```diff
@@ -12,42 +12,93 @@ pragma solidity ^0.8.10;
 
 contract factory{
     //global variables
-
-    address playerA,playerB;
-    uint256 gameState = 0;
+    address playerA;
+    address playerB;
+    bool gameStarted = false;
+    uint turn = 0; //0->game not started, 1-> playera, 2->playerb, 3->gameover
     mapping(address => bool) gameExists;
     mapping(address => uint256) amountStaked;
-
-
+    mapping(uint => address) gameState; //1 to 25
+    uint bomb_tile = 0;
 
     //events
-    event gameCreated( address indexed player1, address indexed amountA  );
+    event gameCreated( address indexed player, uint amount );
+    event gameOver( address indexed player); //player lost
+    event next_player_turn(address player); //turn of player
+    event game_started();
 
-    constructor( ){
+    constructor(){
         playerA = msg.sender;
-        gameState = 
+        gameStarted = false;
+        bomb_tile = returnRandomNumber();
     }
 
-    function createNewGame payable public (){
-        // check the gameExists mapping
-        // set the amountStaked mapping
-        // emit gameCreated event
+    function createNewGame() payable public {
+        require(playerA != address(0));
+        gameStarted = false;
+        gameExists[playerA] =  true;
+        amountStaked[playerA] = msg.value;
+        emit gameCreated(playerA, amountStaked[playerA]);
     }
-
-     
-    function joinGame( address _playerA ){
-        //This function will take address of the player who created the game.
-        //This new player will be able to join this game only if he commits to a amount greater than or equal to what playerA commited
-        require( msg.value => amountStaked[address]);
-        // A mapping to begin the game maybe?
+ 
+    function joinGame( address _playerB ) payable public {
+        playerB = _playerB;
+        require(msg.value >= amountStaked[playerA]);
+        gameExists[playerB] =  true;
+        amountStaked[playerB] = msg.value;
+        emit gameCreated(playerB, amountStaked[playerB]);

```

Comment:
We shouldn't be emitting out events every time a new player joins. The point of creating events when playerA created the game was that our frontend could pick up those events and show them to our ( not yet decided ) playerB